contrib/init.d/pbs_mom.in

```diff
@@ -28,12 +29,30 @@ else
 fi
 
 pidof_pbs_mom() {
-	pid=`pidof pbs_mom`
-	if [ $? -eq 0 ]; then
-		echo $pid
-	else
-		echo "-1"
-	fi
+    pids=`pidof pbs_mom`
+    if [ $? -ne 0 ]; then
+        echo "-1"
+        return 1
+    fi
+
+    lockfile_info=`stat -Lc "%d:%i" $MOM_LOCK 2>/dev/null`
+    if [ $? -ne 0 ]; then
+        echo "-1"
+        return 1
+    fi
+
+    for ((fd=3; fd <= 10; fd++)); do
```

Comment:
so this will work for us, but in theory, there could be even higher fds. you could try that in a second loop where you do the normal/slow approach of loping over all fds of all pids.
also, you could optimise it even further with 
```
for fd in 5 4 3  `6 10`; do
...
done
```